doc/source/colormap.rst

```diff
@@ -1,8 +1,92 @@
-ColorMap
-========
+Color maps
+==========
+
+A color map defines a relationship between scalar data values and a range of colors. 
+Color maps are commonly used to generate false color images, color scatter-plot points, 
+and illustrate the height of surface plots.
+
+PyQtGraph's ColorMap object can also be used a convenient source of colors from a 
+consistent palette or to generate QPen and QBrush objects used to draw lines and fills
+that are colored according to their values along the horizontal or vertical axis.
+
+
+
+Sources for color maps
+----------------------
+
+Color maps can be user defined by assigning a number of *stops* over the range of 0 to 1. A color is given for each stop, and the in-between values are generated by interpolation.
+
+When map colors directly represent values, an improperly designed map can obscure detail over certain ranges of values, while creating false detail in others. PyQtGraph includes the perceptually uniform color maps provided by the `colorcet project` <https://colorcet.holoviz.org/>. Color maps can also be imported from the `colorcet` library or from `Matplotlib`, if either of these is installed.
+
+To see all available color maps, please run the ColorMap demonstration available in the suite of :ref:`examples`.
+
+Examples
```

Comment:
I'm a bit on the fence on this. I'm absolutely in favor of having small usage examples with visuals in the docs, but I think we should be careful about how to do it in a way that's scalable and won't lead to docs that are out of sync with the library. I've been working on an extension to [sphinx-gallery](https://sphinx-gallery.github.io/stable/index.html) which could be used to generate a gallery of the full example scripts. I've also dreamt of a sphinx extension to generate renderings of code blocks in the docs. I suppose we can always leave this in and replace it later if something like that ever gets going. Currently waiting on readthedocs to support installation of apt packages to install xvfb for rendering.

One thing we could do for now is use your scripts for generating these images and just list that code here via [`literalinclude`](https://www.sphinx-doc.org/en/master/usage/restructuredtext/directives.html#directive-literalinclude) and appropriate line numbers. It might be worth converting the code to a more "script-like" structure in that case.